### ISSUE TITLE
[TypeError: Network request failed] Error on RN App

### ISSUE THREAD

**Issue by Yeab13** [facebook/react-native#30677]
Hello,

   My app was working fine earlier but now it stopped responding to API requests. When I check the logs on the API , the requests made from the mobile app didn't make it ( no request received ).

I have cross-checked  like

`android:usesCleartextTraffic="true"` inside `AndroidMainfest.xml`

`<uses-permission android:name="android.permission.INTERNET" />`

This is my API call ( from the mobile app )

`export const getSearches = data =>
  fetch(`${constants.baseURL}searches`, {
    method: 'POST',
    headers: {'Content-Type': 'application/json'},
    body: JSON.stringify(data)
  })
    .then(response => response.json())
    .then(responseData => {
      return responseData;
    })
    .catch(error => {
      console.log('Error......');
      console.warn(error);
    });`

But what I am getting now is the catch block result: `[TypeError: Network request failed]`

My dependencies are : 

`"react": "16.9.0",
  "react-native": "0.61.5"`

The App was working fine but I don't know what was the reason. Any help? ( I tried on Android only )

**Comment by Yeab13** [comment]
Here is my Sys.Info

System:
    OS: macOS 11.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 53.64 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.18.1 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 14.3, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK:
      API Levels: 19, 23, 25, 28, 29, 30
      Build Tools: 27.0.3, 28.0.3, 29.0.2, 29.0.3, 30.0.1
      System Images: android-22 | Google APIs Intel x86 Atom, android-24 | Google Play Intel x86 Atom, android-25 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom, android-29 | Google Play Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6514223
    Xcode: 12.3/12C33 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.5 => 0.61.5 
  npmGlobalPackages:
    react-native-cli: 2.0.1
    react-native-create-library: 3.1.2
    react-native-macos-cli: 2.0.1
    react-native-rename: 2.4.1
    react-native-vector-icons: 7.0.0
    react-native-web: 0.13.1

**Comment by haseebb** [comment]
I am also facing this issue on Hawai p30 lite in android.

**Comment by pedrocardoz0** [comment]
I had the same problem. 

> _But in my case were because my emulator was offline._

Just to make sure that your device is on:

> _You can open the browser and see if it's connected_

**Comment by Yeab13** [comment]
Thank You for your help. Finally, I have discovered the issue. I was using a third party SDK which secures the HTTP communication between App & API. The SDK uses an **SSL Pinning** and it tracks the API server certificate. Last week, my API certificate got updated/renewed by itself, and both of them gets out of sync. The SDK expects the API server's old certificate while the API server has renewed itself and I fixed it by updating the certificate on the SDK solution.
